flake: use makeWrapper to add cardano-cli/-node to PATH

marijanp opened the pull request:
The need to manually ensure that the proper `cardano-cli` and `cardano-node` executables are available in the `PATH` variable is not a very _nixy_ approach. Threfore I've introduced `makeWrapper` to ensure that the proper executables are in the environment.

I did that for `local-cluster` and `plutip-tests` both run successfully now. Please let me know what other executables require to have `cardano-cli` and `cardano-node` in the `PATH`

Editing this flake I also found it very difficult to edit the `cabalProject` function parameters like `modules`. Wrapping `cabalProject` inside another function is a huge anti-pattern in my opinion. As soon as there are more options required you will basically find yourself writing an additional parameter for `nix/haskell.nix` just to propagate it to `cabalProject`

marijanp commented:
Well the readme says that you need to make sure that `cardano-node` and  `cardano-cli` are present in the environment.
[see readme](https://github.com/mlabs-haskell/plutip#requirements)
Moreover when I run the following (master branch)
```
nix run .#"plutip:exe:local-cluster"
```
using the code from `master` i get
```
This processes should be available in the environment:
 ["cardano-node","cardano-cli"]
 but only these were found:
 []
local-cluster: ExitFailure 1
```
And to be honest there is a lot that can go wrong when requiring a user to take care of having the proper executable versions in the environment.

brainrake commented:
Tested, works. Please fix the PR (remove makeWrapper for plutip-tests if not needed) and update readme.

marijanp commented:
@brainrake I've reverted the usage of makeWrapper for plutip-tests. And I've updated the wording. Users of plutip will still need to make sure that `cardano-cli` and `cardano-node` are present if they are depending on the library for their own tests.
Please tell me if that wording is ok.

brainrake commented:
LGTM on my side. Thank you again for your contribution, as well as updates!
@mikekeke please merge

mikekeke commented:
@marijanp
Thank you a lot for contribution!
